Split proto/mod.rs into multiple files

benesch opened the pull request:
This code is pure code movement, with the exception of a few `pub`
modifiers and `use` statements.

Fix #9.

jonhoo commented:
Did anything become `pub` in this, or just `pub(crate)`?

jonhoo commented:
Reading through it, it looks like there are definitely some `pub`s introduced here that should be `pub(crate)` (or maybe even just `pub(super)`?).

benesch commented:
Apologies for the multi-force-pushing; the result is the original commit but rebased on the fix for Travis CI, followed by a commit that addresses review feedback.

> Reading through it, it looks like there are definitely some pubs introduced here that should be pub(crate) (or maybe even just pub(super)?).

Oh, is the rule of thumb that `pub(SCOPE)` should be preferred to `mod internal { pub thing_that_is_actually_semi_private }` wherever possible? Assuming yes, and adjusted the PR as such.

jonhoo commented:
Yes -- in general, it's best if only things that are _actually_ `pub` are marked as such. It makes it much easier to do simple searches and visual scans for what is a part of the public-facing API. `pub(crate)` is generally what you want, though in some very specific cases you might find `pub(super)` or `pub(in p::a::t::h)` handy if you have some internals you really don't want to be used outside of a particular submodule

jonhoo commented on src/proto/active_packetizer.rs:
```diff
@@ -0,0 +1,409 @@
+    timeout: time::Duration,
+
+    /// Bytes we have not yet set.
+    pub outbox: Vec<u8>,
```
I think this should also not be `pub`?

benesch replied:
Good eye! Done.

jonhoo commented:
Awesome, thanks!
